### PR TITLE
Reek excluded_path configuration not being picked up

### PR DESCRIPTION
**Pull request by AllanSiqueira**
## Motivation

When using the key `excluded_files` in `.reek.yml`, the parameter have been ignored in the rubycritic

## Proposal

Looking in the Reek implementation, this filter is applied before he comes in the `Reek::Examiner`, so, a workaround for this issue is to filter the files before sending to Reek::Examiner.

I can add more test scenarios for corner cases if you can give me insights about this flow.

## Check list:
- [X] Add an entry to the [changelog](/CHANGELOG.md)
- [X] [Squash all commits into a single one](/CONTRIBUTING.md)
- [X] Describe your PR, link issues, etc.

Issue: #166

### CODE REVIEW

File: lib/rubycritic/analysers/smells/reek.rb [hunk 1]
```diff
@@ -26,6 +26,8 @@ def to_s
 
       private
 
+      attr_reader :analysed_modules
+
       def add_smells_to(analysed_module)
         Reek.new(analysed_module.pathname).smells.each do |smell|
```
Comment (nunosilva800) [inline lib/rubycritic/analysers/smells/reek.rb]: Maybe by using `Reek::Examiner` instead of the top level module will work?

`examiner = Reek::Examiner.new(source)`

Also, this `examiner` can be memoized.

Comment (AllanSiqueira) [reply lib/rubycritic/analysers/smells/reek.rb]: This is not using the top level module, he is using the `lib/rubycritic/analysers/helpers/reek.rb` module, I tried use the examiner this way, passing the configuration and still dont work

Doing some research on reek's source code I see the `excluded_paths` param is filtered before he reach the `Reek::Examiner`, in [SourceLocator](https://github.com/troessner/reek/blob/master/lib/reek/source/source_locator.rb).
